server/task/src/main/scala/org/apache/james/task/eventsourcing/DecisionProjection.scala

```diff
@@ -18,24 +18,23 @@
   * ***************************************************************/
 package org.apache.james.task.eventsourcing
 
-import org.apache.james.eventsourcing.eventstore.History
+import org.apache.james.eventsourcing.Event
 import org.apache.james.task.TaskManager.Status
 
 case class DecisionProjection(status: Status)
 
 object DecisionProjection {
-  def load(history: History) = {
-    import scala.jdk.CollectionConverters._
-    history.getEvents.asScala
+  def load(events: List[Event]): Option[DecisionProjection] = {
```

Comment:
sorry but it doesn't answer to my comment: a projection has usually not access to the history. Here you don't pass History anymore but cheat by passing the history as a list of Event. I meant you should update the projection for every single event and `fold` on the history to init the projection

Reply:
I think the decision projection should be constructed with the history of events the aggregate is constructed with. Otherwise i see no reason to have an history in the first place.
And it's the only way to be sure the decision projection is in sync with the history.

Reply:
What do you think of the updated version ?